test/oban/testing_test.exs

```diff
@@ -192,6 +192,11 @@ defmodule Oban.TestingTest do
     after
       :telemetry.detach("perform-job-handler")
     end
+
+    test "changing testing mode via opts" do
```

Comment:
An expansion test case that matches my original failure mode would be to `perform_job` on a Job that inserts other Jobs while in `inline` mode, but I wanted to get your take on whether this was too corner-case to persist in the test suite. My original misunderstanding might point towards a doc clarification instead?